simulationdataschema/atoms_state.py

```diff
@@ -0,0 +1,571 @@
+        The electron charge excited for modelling purposes. This is a number between 0 and 1 (Janak state).
+        If `dscf_state` is set to 'initial', then this quantity is set to None (but assumed to be excited
+        to an excited state).
+        """,
+    )
+
+    dscf_state = Quantity(
+        type=MEnum("initial", "final"),
+        description="""
+        Tag used to identify the role in the workflow of the same name. Allowed values are 'initial'
+        (not to be confused with the _initial-state approximation_) and 'final'. If 'initial'
+        is used, then `n_excited_electrons` is set to None and the `orbital_ref.degeneracy` is
+        set to 1.
+        """,
+    )
+
+    def resolve_occupation(self, logger: BoundLogger) -> None:
+        """
+        Resolves the occupation of the orbital state. The occupation is resolved from the degeneracy
+        and the number of excited electrons.
+
+        Args:
+            logger (BoundLogger): The logger to log messages.
+        """
+        if self.orbital_ref is None or self.n_excited_electrons is None:
+            logger.warning(
+                "Cannot resolve occupation without `orbital_ref` or `n_excited_electrons`."
+            )
+            return
+        if self.orbital_ref.occupation is None:
+            degeneracy = self.orbital_ref.resolve_degeneracy()
+            if degeneracy is None:
+                logger.warning("Cannot resolve occupation without `degeneracy`.")
+                return
+            self.orbital_ref.occupation = degeneracy - self.n_excited_electrons
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+        # Check if n_excited_electrons is between 0 and 1
+        if 0.0 <= self.n_excited_electrons <= 1.0:
+            logger.error("Number of excited electrons must be between 0 and 1.")
+
+        # If dscf_state is 'initial', then n_excited_electrons is set to 0
+        if self.dscf_state == "initial":
+            self.n_excited_electrons = None
+            self.degeneracy = 1
+
+        # Resolve the occupation of the active orbital state
+        if self.orbital_ref is not None and self.n_excited_electrons:
+            if self.orbital_ref.occupation is None:
+                self.resolve_occupation(logger)
+
+
+class HubbardInteractions(ArchiveSection):
+    """
+    A base section to define the Hubbard interactions of the system.
+    """
+
+    orbitals_ref = Quantity(
+        type=OrbitalsState,
+        shape=["*"],
+        description="""
+        Reference to the `OrbitalsState` sections that are used as a basis to obtain the Hubbard
+        interaction matrices.
+        """,
+    )
+
+    umn = Quantity(
+        type=np.float64,
+        shape=["*", "*"],
+        unit="joule",
+        description="""
+        Value of the local Hubbard interaction matrix. The order of the rows and columns coincide
+        with the elements in `orbital_ref`.
+        """,
+    )
```

Comment:
I know that this is just a code migration, but I don't remember ever getting to review these extensions to the Hubbard parameters.

My only remark is about the naming. I really had no idea what `umn` was supposed to represent, until I read the description. Maybe you can make it clearer?
Moreover, are there any limitations on the matrix dimensions? Maybe the no. rows = l-degeneracy (at least in most DFT codes)?

Reply:
> Yes, the shape of this matrix coincides with the length of `orbitals_ref`.

Maybe we should consider adding the shapes again? Let's at least keep it in the `description` for now.

Reply:
Ok, I add them back. The checks are not being performed, but this is something we should discuss further.